arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/continuations/result.kt

```diff
@@ -0,0 +1,51 @@
+package arrow.core.continuations
+
+import arrow.core.Either
+import arrow.core.Validated
+import arrow.core.identity
+
+public object ResultEffectScope {
+
+  public fun <A> Result<A>.bind(): A =
+    getOrThrow()
+
+  public fun <A> Either<Throwable, A>.bind(): A =
+    fold({ throw it }, ::identity)
+
+  public fun <A> Validated<Throwable, A>.bind(): A =
+    fold({ throw it }, ::identity)
+}
+
+@Suppress("ClassName")
+public object result {
```

Comment:
If we only have the `inline `version, we probably don't need the `object` indirection here.